### PR TITLE
UserService_AddTestsToValidator

### PR DESCRIPTION
**Pull request by DanilFadeeev**

### CODE REVIEW

**Review by Cerberous007 — COMMENTED** [review]
I'm very glad that you made a new branch that doesn't have extra commits.
But the name is still not good(

File: Services/UserService/Controllers/UsersController.cs [hunk 1]
```diff
@@ -19,6 +21,16 @@ public class UsersController : ControllerBase
         [HttpPost]
         public async Task<bool> CreateUser([FromServices] ICreateUserCommand command, [FromBody] CreateUserRequest request)
```
Comment (dTarasova) [inline Services/UserService/Controllers/UsersController.cs:22]: Please don't write logic in the controller, it is already done in CreateUserCommand

File: Tests/AuthServiceTests/Validators/CreateUserRequestValidator.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
+﻿using DTO.RestRequests;
+using NUnit.Framework;
+using System;
+using UserService.Validators;
+using static UserService.Validators.ErrorsMessages;
+
+
+namespace AuthenticationServiceTests.Validators
```
Comment (dTarasova) [inline Tests/AuthServiceTests/Validators/CreateUserRequestValidator.cs:8]: Why this namespace

Comment (dTarasova) [reply Tests/AuthServiceTests/Validators/CreateUserRequestValidator.cs:8]: And in Auth service

File: Tests/AuthServiceTests/Validators/CreateUserRequestValidator.cs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+{
+    class CreateUserRequestValidatorTests
```
Comment (dTarasova) [inline Tests/AuthServiceTests/Validators/CreateUserRequestValidator.cs:10]: I think we already have the same validator , but it named incorrectly: emailPasswordValidatorTests

**Review by dTarasova — CHANGES_REQUESTED** [review]

File: Tests/AuthServiceTests/Validators/CreateUserRequestValidator.cs [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    {
+        CreateUserRequestValidator validator;
```
Comment (nmamro) [inline Tests/AuthServiceTests/Validators/CreateUserRequestValidator.cs:12]: Public/private/iternal?

File: Services/UserService/Controllers/UsersController.cs [hunk 5]
```diff
@@ -19,6 +21,16 @@ public class UsersController : ControllerBase
         [HttpPost]
         public async Task<bool> CreateUser([FromServices] ICreateUserCommand command, [FromBody] CreateUserRequest request)
         {
+            CreateUserRequestValidator validator = new CreateUserRequestValidator();
```
Comment (nmamro) [inline Services/UserService/Controllers/UsersController.cs:24]: Inject validator, then add from services

File: Services/UserService/Controllers/UsersController.cs [hunk 6]
```diff
@@ -19,6 +21,16 @@ public class UsersController : ControllerBase
         [HttpPost]
         public async Task<bool> CreateUser([FromServices] ICreateUserCommand command, [FromBody] CreateUserRequest request)
         {
+            CreateUserRequestValidator validator = new CreateUserRequestValidator();
+            var validationResult = await validator.ValidateAsync(request);
```
Comment (nmamro) [inline Services/UserService/Controllers/UsersController.cs:25]: There's Validate and throw for these occasions, so mb use it?

File: Tests/AuthServiceTests/Validators/CreateUserRequestValidator.cs [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+        CreateUserRequest correctData;
+        CreateUserRequest incorrectBirthday;
+        CreateUserRequest incorrectEmail;
+        CreateUserRequest incorrectFirstName;
+        CreateUserRequest incorrectLastName;
+        CreateUserRequest emptyPassword;
+
+        [SetUp]
```
Comment (nmamro) [inline Tests/AuthServiceTests/Validators/CreateUserRequestValidator.cs:20]: This is called before each test

**Review by nmamro — CHANGES_REQUESTED** [review]

File: Services/UserService/Controllers/UsersController.cs (hunk 5 above)
Comment (DanilFadeeev) [reply Services/UserService/Controllers/UsersController.cs:24]: why?

**Pull request closed without merging**